src/components/PrimaryPanes/Outline.js

```diff
@@ -39,6 +53,46 @@ export class Outline extends Component<Props> {
     selectLocation({ sourceId: selectedSourceId, line: startLine });
   }
 
+  onContextMenu(event: SyntheticEvent<HTMLElement>, func: SymbolDeclaration) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    const {
+      selectedSource,
+      getFunctionText,
+      getFunctionLocation,
+      flashLineRange,
+      selectedLocation
+    } = this.props;
+
+    const copyFunctionKey = L10N.getStr("copyFunction.accesskey");
+    const copyFunctionLabel = L10N.getStr("copyFunction.label");
+
+    if (!selectedSource) {
+      return;
+    }
+
+    const sourceLine = func.location.start.line;
+    const functionText = getFunctionText(sourceLine);
```

Comment:
We can probably move this inside of click since the user might not always want to copy the function, so we can save some CPU cycles there.

Reply:
Unfortunately no, we need `functionText` to decide wether "copy function" is available

Reply:
But maybe there is better way to check it, this code was copied from `EditorMenu`

Reply:
Empty functions are really rare. So its very least likely that the text will be empty, that's why I suggest we move it inside.

And we can do the same at other places as well.

Reply:
Sounds reasonable for this particular place.
Not sure we should do same optimization in `EditorMenu`